### PR TITLE
Implemented the methods.

### PR DESCRIPTION
**Pull request by KleerK**
I forgot to fork the repo and tried to commit directly to the original, so I had to fork it and clone again...

### CODE REVIEW

**Comment by KleerK** [comment]
by Adrian Tofan.

File: comp110-worksheet-7/DirectoryUtils.cs [hunk 1]
```diff
@@ -7,54 +7,140 @@
...
-		// Get all files whose size is equal to the given value (in bytes) below the given directory
-		public static IEnumerable<string> GetFilesOfSize(string directory, long size)
-		{
-			throw new NotImplementedException();
-		}
-	}
+    public static class DirectoryUtils
+    {
+        public static Tuple<string, long> min_file_size = new Tuple<string, long>(" ", long.MaxValue);
+        public static Tuple<string, long> max_file_size = new Tuple<string, long>(" ", long.MinValue);
```
Comment (edpowley) [inline comp110-worksheet-7/DirectoryUtils.cs:13]: These don't appear to be used anywhere?

File: comp110-worksheet-7/DirectoryUtils.cs [hunk 2]
```diff
@@ -7,54 +7,140 @@
...
+        }
+
+        // Return true if the given path points to a directory, false if it points to a file
+        public static bool IsDirectory(string path)
+        {
+            return File.GetAttributes(path).HasFlag(FileAttributes.Directory);
+        }
+
+        // Method added by me: returns a string array of files in the specified directory by reference.
+        public static void GetFileArray(out string[] file_array, string dir)
```
Comment (edpowley) [inline comp110-worksheet-7/DirectoryUtils.cs:28]: I don't think you need to use an `out` parameter here -- you could just return the array.

File: comp110-worksheet-7/DirectoryUtils.cs [hunk 3]
```diff
@@ -7,54 +7,140 @@
...
+                counter++;
+            return counter;
+
+
+        }
+
+        // Get the path and size (in bytes) of the largest file below the given directory
+        /* We load the array with the files (names and size) and a minimum tuple with no name and min size, then use a query(LINQ) to find the smallest file and then
+         * save its size. A looped / recursive approach of the classic min-finding algorithm could've been also used.
+        */
```
Comment (edpowley) [inline comp110-worksheet-7/DirectoryUtils.cs:86]: This type of information about the implementation of the method is usually better placed in comments within the method itself

File: comp110-worksheet-7/DirectoryUtils.cs [hunk 4]
```diff
@@ -7,54 +7,140 @@
...
+        public static Tuple<string, long> GetSmallestFile(string directory)
+
+        {
+
+
+            string[] file_array;
+            GetFileArray(out file_array, directory);
+            // string[] subdir_array = Directory.GetDirectories(directory);
+            Tuple<string, long> file_size = new Tuple<string, long>(" ", long.MaxValue);
+            string min_file = (from file in file_array let length = GetFileSize(file) where length > 0 orderby length ascending select file).First();
```
Comment (edpowley) [inline comp110-worksheet-7/DirectoryUtils.cs:97]: Nice use of LINQ

**Pull request closed without merging**